viewer/packages/camera-manager/src/ComboControls.ts

```diff
@@ -583,37 +591,58 @@ export default class ComboControls extends EventDispatcher {
     const cameraDirection = reusableVector3;
     let radius = distToTarget + deltaDistance;
 
-    if (radius < minDistance) {
-      radius = minDistance;
-      if (dynamicTarget) {
-        // push targetEnd forward
-        reusableCamera.getWorldDirection(cameraDirection);
-        targetEnd.add(cameraDirection.normalize().multiplyScalar(Math.abs(deltaDistance)));
-      } else {
-        // stops camera from moving forward
-        deltaDistance = distToTarget - radius;
+    if (moveTarget) {
+      // move target together with the camera (for 'w' and 's' keys movement)
+      reusableCamera.getWorldDirection(cameraDirection);
+      targetEnd.add(cameraDirection.normalize().multiplyScalar(-deltaDistance));
+    } else {
+      // behaviour for scrolling with mouse wheel
+      if (radius < minDistance) {
+        radius = minDistance;
+        if (dynamicTarget) {
+          // push targetEnd forward
+          reusableCamera.getWorldDirection(cameraDirection);
+          targetEnd.add(cameraDirection.normalize().multiplyScalar(Math.abs(deltaDistance)));
+        } else {
+          // stops camera from moving forward only if target became close to scroll target
+          if (this.scrollTarget.clone().sub(this.target).length() < 0.2) deltaDistance = distToTarget - radius;
+        }
       }
-    }
 
-    const distFromRayOrigin = -deltaDistance * ratio;
+      // Here we use the law of sines to determine how far we want to move the target.
+      // Direction is always determined by scrollTarget-target vector
+      const tarSTarVec = reusableVector3.subVectors(this.scrollTarget, this.target).normalize(),
+        camTarVec = new Vector3().subVectors(this.target, this.camera.position),
+        camSTarVec = new Vector3().subVectors(this.scrollTarget, this.camera.position);
+
+      const tarCamSTarAngle = camTarVec.angleTo(camSTarVec),
+        tarSTarCamAngle = tarSTarVec.negate().angleTo(camSTarVec.negate());
```

Comment:
Try to find some better names for these variables. Long variable names is ok if needed. Better to be expressive than save storage space for the script.